static/js/stream_edit.js

```diff
@@ -86,11 +104,10 @@ exports.open_edit_panel_empty = function () {
     exports.setup_subscriptions_tab_hash(tab_key);
 };
 
-function format_member_list_elem(email) {
-    const person = people.get_by_email(email);
+function format_member_list_elem(person) {
     return render_stream_member_list_entry({
-        name: person.full_name, email: email,
-        displaying_for_admin: page_params.is_admin,
+        name: person.full_name, email: settings_data.email_for_user_settings(person),
+        displaying_for_admin: page_params.is_admin, show_email: settings_data.show_email(),
```

Comment:
I changed this to wrap 1 entry per line, since that's a lot more readable now that there's more complex code here.